ray-operator/controllers/ray/utils/util.go

```diff
@@ -508,3 +509,27 @@ func FindContainerPort(container *corev1.Container, portName string, defaultPort
 	}
 	return defaultPort
 }
+
+// IsJobFinished checks whether the given Job has finished execution.
```

Comment:
(You can ignore this comment if we decide to merge the logic.)
If we only need to check `batchv1.JobFailed`, we may be able to remove this function.